packages/driver/types/internal-types.d.ts

```diff
@@ -45,6 +45,9 @@ declare namespace Cypress {
     isAnticipatingCrossOriginResponseFor: IStability['isAnticipatingCrossOriginResponseFor']
     fail: (err: Error, options:{ async?: boolean }) => Error
     getRemoteLocation: ILocation['getRemoteLocation']
+    createSnapshot:  ISnapshots['createSnapshot']
+    // NOTE: This command is added in support/utils.js for testing convenience and does not ship with cy!
+    shouldWithTimeout(cb: (subj: {}) => void, timeout?: number): Chainable
```

Comment:
I agree that separating it into a different file is a good idea. 

Also, it should be under `interface Chainable` so it's possible to chain it like `cy.get('div').shouldWithTimeout(...)`.

Reply:
We could, but I think the only two custom commands we are adding to the test suite are `getAll` and `shouldWithTimeout` (at least that I am aware of). Wonder if it would be worth introducing a new pattern since we already have these internal types.

Reply:
What I will do then is create a `spec-types.d.ts` and add `shouldWithTimeout` and `getAll` under chain-able. I think I can do this with just declaring the `Cypress` namespace and adding the `Chainable` interface.

Reply:
added in [ecb6b84](https://github.com/cypress-io/cypress/pull/20949/commits/ecb6b84236953d90502c9414576763bccbdbeec9)